### PR TITLE
chore: make sure directives are correct

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Closes https://github.com/coder/envbuilder/issues/217

### CODE REVIEW

**Review by johnstcn — COMMENTED** [review]
I verified the behaviour is correct 👍 
Do we need to modify these directives if we are not pushing the image?

**Comment by BrunoQuaresma** [comment]
@johnstcn Good question, from what I can see on https://github.com/coder/envbuilder/issues/217 we should not. I'm going to submit a fix.

File: integration/integration_test.go [hunk 1]
```diff
@@ -1354,6 +1354,48 @@ COPY --from=a /root/date.txt /date.txt`, testImageAlpine, testImageAlpine),
...
+	require.NoError(t, err)
+
+	image, err := remote.Image(ref)
+	require.NoError(t, err, "expected image to be present after build + push")
+	configFile, err := image.ConfigFile()
+	require.NoError(t, err, "expected image to return a config file")
+
+	require.Equal(t, configFile.Config.User, "root", "value does not match any of the possible root user values.")
+	require.Equal(t, configFile.Config.WorkingDir, "/", "expected image to have root working directory")
+	require.Equal(t, configFile.Config.Entrypoint, []string{"/.envbuilder/bin/envbuilder"}, "expected image to have envbuilder entrypoint")
```
Comment (johnstcn) [inline integration/integration_test.go:1396]: probably a good idea to assert the opposite in some other test when we don't push

File: envbuilder.go [hunk 2]
```diff
@@ -342,6 +342,14 @@ func Run(ctx context.Context, options Options) error {
 		}
 	}
 
+	if options.PushImage {
+		// Make sure the Dockerfile is using the correct directives to run envbuilder
+		buildParams.DockerfileContent = buildParams.DockerfileContent + "\n" +
```
Comment (mtojek) [inline envbuilder.go:347]: I guess that you'll have a merge conflict with the other PR: https://github.com/coder/envbuilder/issues/216

File: integration/integration_test.go [hunk 3]
```diff
@@ -1354,6 +1354,48 @@ COPY --from=a /root/date.txt /date.txt`, testImageAlpine, testImageAlpine),
...
+	t.Parallel()
+
+	srv := createGitServer(t, gitServerOptions{
+		files: map[string]string{
+			".devcontainer/entrypoint.sh": "#!/bin/sh\necho 'Hello, world!'",
+			".devcontainer/Dockerfile": "FROM " + testImageAlpine + "\n" +
+				"RUN addgroup -S coder && adduser -S coder -G coder \n" +
+				"USER coder\n" +
+				"WORKDIR /app\n" +
+				"ENTRYPOINT [\"./entrypoint.sh\"]",
```
Comment (mtojek) [inline integration/integration_test.go:1367]: maybe add a comment here that these settings should be overridden by envbuilder

File: integration/integration_test.go (hunk 1 above)
Comment (BrunoQuaresma) [reply integration/integration_test.go:1396]: If they are not pushed, what reference can we use to get the local image and inspect it?

**Comment by BrunoQuaresma** [comment]
This is blocked by https://github.com/coder/envbuilder/issues/216 because it needs to have the embed binary to set the entrypoint correctly

**Comment by johnstcn** [comment]
closing in favour of https://github.com/coder/envbuilder/pull/246

**Pull request closed without merging**